pandas/indexes/base.py

```diff
@@ -2886,6 +2888,9 @@ def join(self, other, how='left', level=None, return_indexers=False):
         elif how == 'outer':
             join_index = self.union(other)
 
+        if sort:
```

Comment:
In file `tests/frame/test_join.py` I have created 2 methods `test_join` and `test_join_sort` where I test the joined result for each of the 4 values of `how`, for default `sort=False` and `sort=True`, respectively. Is that OK for you? Or would you need further examples?